staker-rewards/test/sla-calculator.test.js

```diff
@@ -0,0 +1,249 @@
+import chai from "chai"
+import testCache from "./data/test-cache.json"
+import SLACalculator from "../lib/sla-calculator.js"
+
+const { assert } = chai
+
+const operator = "0xF1De9490Bf7298b5F350cE74332Ad7cf8d5cB181"
+
+const mockCache = {
+  getKeeps: (status) =>
+    testCache.keeps.filter((keep) => !status || keep.status.name === status),
+}
+
+const createMockContext = () => ({
+  cache: mockCache,
+  contracts: {},
+})
+
+describe("SLA calculator", async () => {
+  it(
+    "should return keygen SLA equal to 100% " +
+      "if operator has no failed keygens",
+    async () => {
+      // The operator has `2` keeps created within this interval. Both keeps
+      // are currently active. This means `2` key generations occurred and
+      // all of them completed successfully.
+      const interval = {
+        start: 9000,
+        end: 20000,
+      }
+
+      const mockContext = createMockContext()
+
+      // Mock the `BondedECDSAKeep` just to make this test pass as it is not
+      // relevant in the context of this scenario.
+      mockContext.contracts.BondedECDSAKeep = {
+        at: () => ({
+          getPastEvents: () => [],
+        }),
+      }
```

Comment:
We have two cases: we either mock `getPastEvents` call to return `[]` or `[{}]`. Given that the way how mocking is set up is quite tricky, maybe we can have two functions declared next to `createMockContext` that would accept `context` and mock those scenarios? For example `recordNoSignatureRequests` and `recordSignatureRequestsForAllKeeps`?

Reply:
I will leave the naming to you, it is not really record-replay mocking, heh.